packages/odata-service-inquirer/src/prompts/datasources/sap-system/system-selection/prompt-helpers.ts

```diff
@@ -157,14 +163,51 @@ export async function connectWithDestination(
  */
 export function getBackendSystemDisplayName(system: BackendSystem): string {
     const userDisplayName = system.userDisplayName ? ` [${system.userDisplayName}]` : '';
-    let systemTypeName = '';
-    if (system.authenticationType === 'reentranceTicket') {
-        systemTypeName = ` (${t('texts.systemTypeS4HC')})`;
+    const systemTypeName = getBackendSystemTypeName(system.systemType);
+    return `${system.name}${systemTypeName}${userDisplayName}`;
+}
+
+/**
+ * Returns the formatted system type name for the given backend system.
+ *
+ * @param systemType the system type to get the name for
+ * @returns system type name formatted as a string, e.g. " (BTP)" or " (S4HC)".
+ */
+function getBackendSystemTypeName(systemType?: string): string {
+    let systemTypeName = ''; // for on prem we do not show the system type
+    if (systemType === 'BTP' || systemType === 'S4HC') {
+        systemTypeName = ` (${systemType})`;
```

Comment:
```suggestion
        systemTypeName = systemType === 'S4HC'` ? t('texts.systemTypeS4HC'} : t('texts.systemTypeBTP'};
```

Reply:
Not necessary?

Reply:
Shouldnt system type be a string type literal which can be translated into an end user system type label. Sure, at the moment, the system type and label are the same, but in the past the label was something else and so decided to externalise it to isolate changes.

Reply:
I've reverted to use the labels again